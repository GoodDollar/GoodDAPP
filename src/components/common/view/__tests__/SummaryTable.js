import React from 'react'
import renderer from 'react-test-renderer'
import ImportedSummaryTable from '../SummaryTable'
import { withThemeProvider } from '../../../../__tests__/__util__'
const SummaryTable = withThemeProvider(ImportedSummaryTable)

// Note: test renderer must be required after react-native.

describe('SummaryTable', () => {
  it('matches snapshot', async () => {
    let component

    await renderer.act(
      // eslint-disable-next-line require-await
      async () =>
        (component = renderer.create(<SummaryTable counterPartyDisplayName="name" amount={20} reason="MyReason" />)),
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with minimal properties', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<SummaryTable amount={20} />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
