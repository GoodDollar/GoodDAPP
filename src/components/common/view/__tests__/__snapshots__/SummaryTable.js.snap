// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryTable matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1oszu61 r-flexDirection-eqz5dr"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitBoxPack": "center",
        "WebkitFlexGrow": 1,
        "WebkitJustifyContent": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "msFlexPack": "center",
        "msFlexPositive": 1,
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-obd0qt r-borderBottomColor-c7jyll r-borderBottomWidth-qklmqi r-borderStyle-1phboty r-flexDirection-18u37iz r-height-eu3ka r-justifyContent-1wtj0ep r-paddingBottom-iphfwy"
      style={
        Object {
          "marginTop": "16px",
        }
      }
    >
      <div
        className="css-text-901oao r-color-vaa0vc r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        To:
      </div>
      <div
        className="css-text-901oao r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-hiw28u r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        name
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-obd0qt r-borderBottomColor-c7jyll r-borderBottomWidth-qklmqi r-borderStyle-1phboty r-flexDirection-18u37iz r-height-eu3ka r-justifyContent-1wtj0ep r-paddingBottom-iphfwy"
      style={
        Object {
          "marginTop": "16px",
        }
      }
    >
      <div
        className="css-text-901oao r-color-vaa0vc r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Amount:
      </div>
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz"
      >
        <div
          className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-marginRight-a5pmau r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
          dir="auto"
        >
          0.20
        </div>
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
        >
          <div
            className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
            dir="auto"
          >
            G
          </div>
          <div
            className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
            dir="auto"
          >
            $
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-obd0qt r-borderBottomColor-c7jyll r-borderBottomWidth-qklmqi r-borderStyle-1phboty r-flexDirection-18u37iz r-height-eu3ka r-justifyContent-1wtj0ep r-paddingBottom-iphfwy"
      style={
        Object {
          "marginTop": "16px",
        }
      }
    >
      <div
        className="css-text-901oao r-color-vaa0vc r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        For:
      </div>
      <div
        className="css-text-901oao css-textMultiLine-cens5h r-color-cddv6f r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-marginLeft-1n0xq6e r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
        style={
          Object {
            "WebkitLineClamp": 2,
          }
        }
      >
        MyReason
      </div>
    </div>
  </div>
</div>
`;

exports[`SummaryTable matches snapshot with minimal properties 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className="css-view-1dbjc4n r-alignItems-1oszu61 r-flexDirection-eqz5dr"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitBoxPack": "center",
        "WebkitFlexGrow": 1,
        "WebkitJustifyContent": "center",
        "flexGrow": 1,
        "justifyContent": "center",
        "msFlexPack": "center",
        "msFlexPositive": 1,
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-obd0qt r-borderBottomColor-c7jyll r-borderBottomWidth-qklmqi r-borderStyle-1phboty r-flexDirection-18u37iz r-height-eu3ka r-justifyContent-1wtj0ep r-paddingBottom-iphfwy"
      style={
        Object {
          "marginTop": "16px",
        }
      }
    >
      <div
        className="css-text-901oao r-color-vaa0vc r-fontFamily-vnw8o6 r-fontSize-1b43r93 r-fontWeight-1it3c9n r-lineHeight-qvk6io r-textAlign-q4m81j r-textDecorationLine-13wfysu r-textTransform-3twk1y"
        dir="auto"
      >
        Amount:
      </div>
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz"
      >
        <div
          className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-hiw28u r-fontWeight-b88u0q r-lineHeight-qvk6io r-marginRight-a5pmau r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
          dir="auto"
        >
          0.20
        </div>
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-justifyContent-1wtj0ep"
        >
          <div
            className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
            dir="auto"
          >
            G
          </div>
          <div
            className="css-text-901oao r-color-1q5uqvh r-fontFamily-6jnb14 r-fontSize-1b43r93 r-fontWeight-b88u0q r-lineHeight-qvk6io r-textAlign-1ff274t r-textDecorationLine-13wfysu r-textTransform-3twk1y"
            dir="auto"
          >
            $
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
